Document which service types are supported, and what they means (#106)
stackabletech/listener-operator: docs/modules/listener-operator/pages/listenerclass.adoc

----
include::example$listenerclass-internal-gke.yaml[]
----

== Service types

The service type is defined by `ListenerClass.spec.serviceType`. The following service types are currently supported by the Stackable Listener Operator:

[#servicetype-clusterip]
=== `ClusterIP`

> The listener can be accessed from inside of the Kubernetes cluster. The listener addresses will direct clients to the cluster-internal address.

Reviewer: ```suggestion
The Listener can be accessed from inside the Kubernetes cluster. The Listener addresses will direct clients to the cluster-internal address.
```

[#servicetype-nodeport]
=== `NodePort`

> The listener can be accessed from outside the Kubernetes cluster. The listener address will direct clients to connect to a randomĺy assigned port on the nodes running the pods.

Reviewer: ```suggestion
The Listener can be accessed from outside the Kubernetes cluster if the nodes have public IPs. The Listener address will direct clients to connect to a randomly assigned port on the nodes running the pods.
```

> Additionally, pods bound to `NodePort` listeners will be xref:volume.adoc#pinning[pinned] to a specific node. If this is undesirable, consider using xref:#servicetype-loadbalancer[] instead.

Reviewer: ```suggestion
Additionally, pods bound to `NodePort` Listeners will be xref:volume.adoc#pinning[pinned] to a specific node. If this is undesirable, consider using xref:#servicetype-loadbalancer[] instead.
```

[#servicetype-loadbalancer]
=== `LoadBalancer`

> The listener can be accessed from outside the Kubernetes cluster. A dedicated address will be allocated for the listener, which any clients will be directed to connect to using.

Reviewer: ```suggestion
The Listener can be accessed from outside the Kubernetes cluster. A dedicated address will be allocated for the Listener.
```
